app/lib/m2m.js

```diff
@@ -48,7 +48,7 @@ async function AddMachineUser(userinfo)
     displayName: userinfo.displayName || "MachineClient",
     emails: [userinfo.email || "unknown@unknown.edu"],
     permissions: userinfo.permissions 
-                  || [ 'components:view', 'components:edit', 'tests:submit', 'tests:view', 'jobs:submit', 'jobs:view', 'forms:edit'],
+                  || [ 'components:view', 'components:edit', 'actions:view', 'actions:perform', 'forms:edit'],
```

Comment:
```suggestion
    || ['components:view', 'components:edit', 'actions:view', 'actions:perform', 'forms:edit'],
```
Spacing in this file needs to be corrected as well

Reply:
Agreed - will deal with it when I look at the m2m side of things.  As with the other file above, for this PR I was just making the minimal required changes.

Reply:
One more thing — `unknown@unknown.edu` should not be used.  It should probably be undefined else null.  If type matters, empty string.  Otherwise, `example.com` is a safe, standardized generic domain used for example purposes.

Reply:
Noted for future reference, thanks.